docs/fxa-engineering/fxa-webchannel-protocol.md

```diff
@@ -0,0 +1,324 @@
+  uid: <user id>
+}
+```
+
+- `email` - User's email address.
+- `uid` - The user identifier.
+
+#### fxaccounts:delete
+
+Sent after the user successfully deletes their account. No response is expected.
+Expected by Fx Desktop.
+
+##### data
+
+```js
+{
+  email: <email>,
+  uid: <user id>
+}
+```
+
+- `email` - User's email address.
+- `uid` - The user identifier.
+
+#### fxaccount:change_password
+
+Sent after the user successfully changes their password. No response is expected. A change password causes a user's `unwrapBKey` to change, all data that is sent with `fxaccounts:login` is sent with `fxaccounts:change_password`.
+
+##### data
+
+See [Login Data](#logindata).
+
+#### fxaccounts:logout
+
+Called whenever the user signs out of their account. No response is expected.
+
+##### data
+
+```js
+{
+  uid: <user id>
+}
+```
+
+- `uid` - The user identifier.
+
+#### profile:change
+
+Sent after the user changes their avatar or display name. No response is
+expected.
+
+##### data
+
+```js
+{
+  uid: <user id>
+}
+```
+
+- `uid` - The user identifier.
+
+### Command data values
+
+#### loginData
+
+```js
+{
+  customizeSync: (true|false),
+  declinedSyncEngines: [<array of declined sync engines>],
+  email: <email>,
+  keyFetchToken: <key fetch token>,
+  offeredSyncEngines: [<array of displayed sync engines>],
+  sessionToken: <session token>,
+  uid: <user id>,
+  unwrapBKey: <unwrap b key>,
+  verified: (true|false),
+  verifiedCanLinkAccount: (true|false)
+}
+```
+
+- `customizeSync` - Should the user be shown the browser's `Customize Sync` dialog?
+- `declinedSyncEngines` - An array of Sync engines the user has deselected. See [declinedSyncEngines, offeredSyncEngines](#declinedsyncengines-offeredsyncengines).
+- `email` - User's email address.
+- `keyFetchToken` - Token used to fetch wrapWrapKB from the server.
+- `offeredSyncEngines` - An array of Sync engines that were displayed to the user. See [declinedSyncEngines, offeredSyncEngines](#declinedsyncengines-offeredsyncengines).
+- `sessionToken` - The current session token that can be used to interact with FxA's auth server.
+- `uid` - The user identifier.
+- `unwrapBKey` - Used to unwrap wrapWrapKB fetched from the server, resulting key is called kB. Keys derived from kB are used for data encryption for Sync, Send, and Notes.
+- `verified` - Is the user verified?
+- `verifiedCanLinkAccount` - Has the user verified they want to link accounts, if signing into a different account than previously signed in to?
+
+#### capabilities
+
+An object containing browser supported capabilities. Only available with browser support.
+
+```js
+{
+  engines: [<list of optional supported engines>],
+  choose_what_to_sync : <boolean>[default: false],
+  pairing: <boolean>
+}
+```
+
+- `engines` - A list of optional supported engines.
+- `choose_what_to_sync` - (OAuth WebChannel Only) - Whether to show the Choose What to Sync screen
+- `pairing` - Whether the browser can act as a pairing authority.
```

Comment:
IIUC we also need to add `multiService` in this list, right?

Reply:
Added.